remote/src/main/java/com/spotify/metrics/remote/OkRemote.java

```diff
@@ -89,4 +91,17 @@ public void onResponse(Call call, Response response) throws IOException {
         });
         return result;
     }
+
+    @Override
+    public void waitForAllCalls() {
+        Dispatcher dispatcher = client.dispatcher();
+        while (dispatcher.queuedCallsCount() != 0 || dispatcher.runningCallsCount() != 0) {
+            try {
+                Thread.sleep(100);
+            } catch (InterruptedException e) {
```

Comment:
Am I missing something here? All the methods are synchronized. Shouldn't the size call be safe in this case?

I'll propagate the exception.